src/tests/SFA.DAS.EmployerIncentives.Commands.UnitTests/UpdateVrfCaseStatusForLegalEntity/Handlers/WhenHandlingUpdateVrfCaseDetailsForLegalEntityCommand.cs

```diff
@@ -0,0 +1,110 @@
+                _fixture.Create<DateTime>());
+
+            var accounts = _fixture.CreateMany<Account>(3).ToList();
+            var legalEntityToBeUpdated = LegalEntity.New(123, _fixture.Create<string>(), LegalEntityToBeUpdatedId);
+
+            accounts[0].AddLegalEntity(1, legalEntityToBeUpdated);
+            accounts[0].AddLegalEntity(2, _fixture.Create<LegalEntity>());
+            accounts[0].AddLegalEntity(3, _fixture.Create<LegalEntity>());
+
+            accounts[1].AddLegalEntity(4, _fixture.Create<LegalEntity>());
+            accounts[1].AddLegalEntity(5, legalEntityToBeUpdated);
+
+            _mockDomainRepository.Setup(x => x.GetByHashedLegalEntityId(command.HashedLegalEntityId)).ReturnsAsync(accounts);
+
+            // Act
+            await _sut.Handle(command);
+
+            // Assert
+            var updatedLegalEntities = accounts.SelectMany(x => x.LegalEntities.Where(e => e.HashedLegalEntityId == command.HashedLegalEntityId)).ToList();
+            updatedLegalEntities.Should().NotBeEmpty();
+            foreach (var legalEntity in updatedLegalEntities)
+            {
+                legalEntity.VrfCaseId.Should().Be(command.CaseId);
+                legalEntity.VrfVendorId.Should().Be(command.VendorId);
+                legalEntity.VrfCaseStatus.Should().Be(command.Status);
+                legalEntity.VrfCaseStatusLastUpdatedDateTime.Should().Be(command.CaseStatusLastUpdatedDate);
+            }
+
+            _mockDomainRepository.Verify(m => m.Save(accounts[0]), Times.Once);
+            _mockDomainRepository.Verify(m => m.Save(accounts[1]), Times.Once);
+            _mockDomainRepository.VerifyAll();
+        }
+
+
+        [Test]
+        public async Task Then_VRF_details_for_given_legal_entity_are_not_updated_if_status_is_completed()
+        {
+            // Arrange
+            var caseCompletedCommand = new UpdateVendorRegistrationCaseStatusCommand(LegalEntityToBeUpdatedId, _fixture.Create<string>(), _fixture.Create<string>(), "case request completed",
+                _fixture.Create<DateTime>());
+
+            var accounts = _fixture.CreateMany<Account>(3).ToList();
+            var legalEntityToBeUpdated = LegalEntity.New(123, _fixture.Create<string>(), LegalEntityToBeUpdatedId);
+
+            accounts[0].AddLegalEntity(1, legalEntityToBeUpdated);
+            accounts[0].AddLegalEntity(2, _fixture.Create<LegalEntity>());
+            accounts[0].AddLegalEntity(3, _fixture.Create<LegalEntity>());
+
+            accounts[1].AddLegalEntity(4, _fixture.Create<LegalEntity>());
+            accounts[1].AddLegalEntity(5, legalEntityToBeUpdated);
+
+            _mockDomainRepository.Setup(x => x.GetByHashedLegalEntityId(caseCompletedCommand.HashedLegalEntityId)).ReturnsAsync(accounts);
+            await _sut.Handle(caseCompletedCommand);
+
```

Comment:
This is proving that after the status update to "Case request complete" we can no longer update the record. I think this is incorrect.